src/compiler/crystal/tools/doc/html/main.html

```diff
@@ -5,8 +5,11 @@
   <meta id="repository-name" content="<%= repository_name %>">
   <title>README - <%= repository_name %></title>
   <script type="text/javascript">
-  CrystalDoc.base_path = "";
+    CrystalDoc.base_path = "";
   </script>
+  <% if repository_name == CRYSTAL_REPOSITORY_NAME %>
+    <link rel="canonical" href="<%= CANONICAL_BASE_URL %>" />
+  <% end %>
```

Comment:
Why not add this to `HeadTemplate`?
src/compiler/crystal/tools/doc/html/_head.html

Reply:
AFAIK there's no `type` in `HeadTemplate`.